data-prepper-plugins/aggregate-processor/src/main/java/com/amazon/dataprepper/plugins/processor/aggregate/AggregateIdentificationKeysHasher.java

```diff
@@ -0,0 +1,26 @@
+/*
+ * Copyright OpenSearch Contributors
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+package com.amazon.dataprepper.plugins.processor.aggregate;
+
+import com.amazon.dataprepper.model.event.Event;
+
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+
+class AggregateIdentificationKeysHasher {
+    AggregateIdentificationKeysHasher() {
+
+    }
+
+    Map<Object, Object> createIdentificationKeyHashFromEvent(final Event event, final List<String> identificationKeys) {
```

Comment:
The `identificationKeys` never changes for any given `AggregateProcessor`. You could make `identificationKeys` a constructor parameter in `AggregateIdentificationKeysHasher`. Then you can remove `identificationKeys` as a parameter from this function.